share/dkg/dkg.go

```diff
@@ -722,6 +735,12 @@ func (d *DistKeyGenerator) ProcessJustifications(bundles []*JustificationBundle)
 		return nil, fmt.Errorf("node can only process justifications after processing responses - current state %s", d.state.String())
 	}
 
+	defer func() {
```

Comment:
This works, but it might be clearer for future readers to do something like:

```
func (d *DistKeyGenerator) ProcessJustifications(bundles []*JustificationBundle) (*Result, error) {
	resp, err := d.processJustifications(bundles)
	if err == nil {
		return nil, d.checkIfEvicted(JustifPhase)
	}
	return resp, err
}

func (d *DistKeyGenerator) processJustifications(bundles []*JustificationBundle) (*Result, error) {
...
```

Reply:
Is it "do it this way it's better" or "that also might be possible but let's not block on it" ? :D

Reply:
I kinda disagree it's better (defer are here exactly for this purpose) so a big "Thank you" 😁

Reply:
My understanding was that defer was there for resource cleanup and handling panics, not misdirecting control flow.

Besides, in this function the only return that can have a `nil` error is the very last one `return d.computeResult()` (as the others return before the defer).

Additionally, I'd question why the named parameters are here as they aren't used - the `res *Result` doesn't get returned at all and the `err error` only gets used by the defer (which is unnecessary) and when it's set in `err = fmt.Errorf("process-justifications: only %d/%d valid deals - dkg abort", allGood, targetThreshold)` which would be clearer again if you just returned `nil, fmt.Errorf("process-justifications: only %d/%d valid deals - dkg abort", allGood, targetThreshold)`

Reply:
Ok I've found a good compromise I think given what you wrote just above. Wdyt ? @CluEleSsUK

Reply:
agree, this looks clear